### ISSUE TITLE
In zimit2 ZIMs, pages that do not have an overflow cannot swipe away the menu and nav bar

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#1266]
I noticed this in the fas-military-medicine zimit2 ZIM. The landing page is too short to have an overflow / scrollbar. In other ZIM types (e.g. PhET) we can still swipe away the menu and navbar, which can otherwise obscure some content at the bottom of the page, but in this Zimit2 ZIM it's not working. This is a minor bug, but as we have code specifically to do this, it's strange that it's not working for this zimit2 ZIM.

**Comment by HUN-sp** [comment]
@audiodude I would like to work on this issue.

**Comment by Jaifroid** [comment]
@HUN-sp Could you kindly explain what solution you have in mind for this issue? Please also take a look at CONTRIBUTING.md in the root to get an idea of what is involved in contributing to this repo. Many thanks!

**Comment by HUN-sp** [comment]
I m looking into it .Will get back to you once done with  a solution which
actually solved

**Comment by Jaifroid** [comment]
Thank you @HUN-sp . It would be good to discuss your idea here before you start any serious work on it, because the swiping code is a bit intricate, and we would need an agreed approach. Currently the swiping away of menus relies on being able to scroll a page. A different solution is needed when the page won't scroll. Just so you know, this issue is not trivial IMHO.

**Comment by VikramAditya33** [comment]
Hi @Jaifroid 
I’d like to look into this issue and try reproducing it locally.
From what I understand, in zimit2 ZIMs (like fas-military-medicine), the swipe-to-hide gesture for the menu and navbar doesn’t trigger on pages without overflow, possibly because the swipe logic depends on scroll or overflow detection.

I’ll start by testing this behavior with that ZIM and review how the swipe gesture is handled in the code for non-scrollable pages.

Will update here once I have some findings or a possible fix.

**Comment by Jaifroid** [comment]
@VikramAditya33 The keyboard Ctrl-UpArrow does work to hide or show the navbar, so you could take a look at the code for that and see why gestures don't work. Some Zimit archives have a separate replay_iframe, but only older ones. Newer Zimit archives don't have that, and really the gestures should work fine, though I haven't tested, as I'd need a non-scrollable page to test.

Be sure to look at CONTRIBUTING to see how to set up, and do please come back here to explain your suggested approach before starting any coding.

**Comment by VikramAditya33** [comment]
Thanks @Jaifroid 
I’ll check how the `Ctrl + UpArrow` keyboard shortcut triggers the navbar hide/show behavior and compare that logic with the gesture handling part.

My initial thought is that in Zimit2 pages without scroll, the swipe detection might be tied to `touchmove` or scroll-based conditions, so I’ll explore making the gesture trigger independent of scroll height.

Once I confirm where that’s handled, I’ll share a short plan here before coding.

**P.S. I've started investigating the code and found some interesting details:**

1. **Keyboard shortcuts work correctly** - Both `Ctrl+UpArrow` and `Ctrl+DownArrow` toggle the navbar (this is intentional design, not a bug)

1. **Gesture issue confirmed** - The touch handler in `uiUtil.js` requires a swipe distance of 1.5x the window height `(delta > articleContainer.contentWindow.innerHeight / 1.5)`, which is impossible to achieve on short pages

3. **Root cause identified** - The keyboard handler has no distance requirement and works immediately, while the touch handler fails on short pages due to the distance threshold

4. **Additional issue discovered** - The forward/back buttons use `history.back()` and `history.forward()`, but since Kiwix-js doesn't maintain browser history for ZIM navigation, these buttons don't work properly (back shows blank page, forward does nothing)

**My proposed approach:** Instead of removing the distance requirement entirely, I could reduce it to a minimal threshold (e.g., 30–50px) to avoid accidental triggers while keeping the gesture functional on short pages. This would make the touch handler work more like the keyboard handler while maintaining good user experience.
Would love to hear your thoughts before proceeding!

**Comment by Jaifroid** [comment]
@VikramAditya33 Thanks very much, it sounds like you've got a good handle on it! That swipe distance sounds like something a bit rough-and-ready indeed, though note it's the innerHeight divided by 1.5, not innerHeight * 1.5, just looking at what you wrote (I didn't check the code). Maybe you can find a fixed sensible value.

I'm assigning the issue to you. Please feel free to make a PR. Read CONTRIBUTING as there are some gotchas in the setup for developing, because the app caches its own code, which means you won't always see changes reflected immediately when testing in browser unless the Bypass AppCache option in Config is on and DevTools is open. If you don't see changes, using Ctrl+Shift+R with DevTools open usually realoads latest code. Sometimes you also have to turn off browser caching in the Network tab of DevTools.

On the history issue, Kiwix JS **does** maintain browser history in most contexts. For example in a Wikipedia ZIM those back/forward buttons should work fine. There is a function in app.js dedicated to storing and retrieving browser history. In some Zimit archives, navigation may not be handled in a way that evokes that algorithm. It's also the case that if you have only visited a single page, the history may not work as expected, which is a bug (sort of). However, I'd say correcting the history handling is a separate issue (feel free to open one if you have a reproducible bug).
